pkg/util/users/users.go

```diff
@@ -120,6 +119,13 @@ func (strategy DefaultUserSyncStrategy) ExecuteSyncRequests(requests []spec.PgSy
 			if err := strategy.alterPgUser(request.User, db); err != nil {
 				reqretries = append(reqretries, request)
 				errors = append(errors, fmt.Sprintf("could not alter user %q: %v", request.User.Name, err))
+				// check if additional owners are misconfigured as members to a database owner (check #1862 for details)
+				// resolve it by revoking the database owner from the additional owner role
+				if request.User.IsDbOwner && len(strategy.AdditionalOwnerRoles) > 0 {
```

Comment:
I think the condition mismatches the comment: the `if` condition is more general.
What the current condition says and the code in the `true` branch does:

```
 Every time we failed (`err != nil` at line 119) to modify a db owner role (`if request.User.IsDbOwner`)
 and there are additional roles defined (`&& len(strategy.AdditionalOwnerRoles) > 0`), revoke all additional roles (lines 163-166) from the db owner.
```
e.g. if a Db is transiently not accessible for a second, we might get some future unrelated additional role removed for no reason.